### PR TITLE
MRCM: Ensure refset range constraints are properly evaluated

### PR DESCRIPTION
**Pull request by AAAlinaaa**

### CODE REVIEW

**Review by cmark — APPROVED** [review]
🥓

File: snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule_mrcm_constraint.groovy [hunk 1]
```diff
@@ -190,7 +190,7 @@ for (String typeId : allowedRanges.keySet()) {
 			String ancestorId = clause.replaceAll("<", "").strip();
 			ancestorIds.add(ancestorId);
 		} else if (clause.startsWith("^")) {
-			String refsetId = clause.replaceAll("^", "").strip();
+			String refsetId = clause.replace("^", "").strip();
```
Comment (nagyo) [inline snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule_mrcm_constraint.groovy:193]: Looks OK, but maybe using a `Charmatcher.is('^').trimLeadingFrom(clause)` would be a more robust solution for all of these string manipulations. According to the javadoc of the Pattern class, there are some special cases where `<` and `!` could fall into the special character category, which could also be interpreted as a Java regex expression.

**Review by nagyo — APPROVED** [review]
LGTM! with a note

File: snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule_mrcm_constraint.groovy (hunk 1 above)
Comment (AAAlinaaa) [reply snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule_mrcm_constraint.groovy:193]: Groovy's default is method overwrites the CharMatcher's is and the result would be a boolean not a CharMatcher.

Comment (nagyo) [reply snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule_mrcm_constraint.groovy:193]: There is `anyOf()` as well

**Pull request merged**